pkg/kubefed2/federate/util.go

```diff
@@ -101,29 +103,37 @@ func LookupAPIResource(config *rest.Config, key, targetVersion string) (*metav1.
 			continue
 		}
 		for _, resource := range resourceList.APIResources {
+			var matchResource = func() {
+				if targetResource != nil {
+					multiMatched = true
+				} else {
+					targetResource = resource.DeepCopy()
+					targetResource.Group = gv.Group
+					targetResource.Version = gv.Version
+				}
+
+				matchedResources = append(matchedResources, groupQualifiedName(resource.Name, gv.Group))
+			}
+
 			if lowerKey == resource.Name ||
 				lowerKey == resource.SingularName ||
 				lowerKey == strings.ToLower(resource.Kind) ||
 				lowerKey == fmt.Sprintf("%s.%s", resource.Name, gv.Group) {
 
-				targetResource = &resource
-				break
+				matchResource()
+				continue
 			}
 			for _, shortName := range resource.ShortNames {
 				if lowerKey == strings.ToLower(shortName) {
-					targetResource = &resource
+					matchResource()
 					break
 				}
 			}
-			if targetResource != nil {
-				break
-			}
-		}
-		if targetResource != nil {
-			targetResource.Group = gv.Group
-			targetResource.Version = gv.Version
-			break
 		}
+
+	}
+	if multiMatched {
+		return nil, errors.Errorf("Multiple resources are matched: %s. A group-qualified plural name must be provided.", strings.Join(matchedResources, ", "))
```

Comment:
Consider indicating the name that was supplied in the error message e.g. `Multiple resources are matched by "<supplied name>":`